libs/api/ebpf_api.cpp

```diff
@@ -161,7 +184,14 @@ create_map_function(
 static map_cache_t&
 get_map_cache_entry(uint64_t map_fd)
 {
-    return _map_file_descriptors[map_fd - 1];
+    size_t size = _map_file_descriptors.size();
+    for (size_t i = 0; i < size; i++) {
+        if (_map_file_descriptors[i].ebpf_map_descriptor.original_fd == map_fd) {
+            return _map_file_descriptors[i];
+        }
+    }
```

Comment:
this would be more efficient if _map_file_descriptors was a std::map with original_fd as the key
i.e. change std::vector<map_cache_t> to std::map<uint64_t, map_cache_t>.  That also protects against having two different entries in the vector with the same original_fd

Reply:
Looks like this is a bit involved change. Created issue #211 to track this. Will work this once this PR is pushed.